[Shivam Tiwari] iP

Shivlock221b opened the pull request:
DukePro
>**_“Your mind is for having ideas, not holding them.” – David Allen (source)_**

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* FAST SUPER FAST to use
All you need to do is,

1. download it from here.
1. double-click it.
1. add your tasks.
let it manage your tasks for you 😉
And it is _FREE!_

Features:

 - [X] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)
If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Shivlock221b committed:
first commit

Shivlock221b committed:
second commit

Shivlock221b committed:
Third commit

Shivlock221b committed:
fourth commit

Shivlock221b committed:
Delete Deadline.class

Shivlock221b committed:
Delete Duke.class

Shivlock221b committed:
Delete Event.class

Shivlock221b committed:
Delete Task.class

Shivlock221b committed:
Delete Todo.class

Shivlock221b committed:
commit test

Shivlock221b committed:
fifth commit

Shivlock221b committed:
sixth commit

Shivlock221b committed:
remove class

Shivlock221b committed:
testing UI

Shivlock221b committed:
testing UI

Shivlock221b committed:
testing tag

Shivlock221b committed:
Level-1

Shivlock221b committed:
Level-1

Shivlock221b committed:
Level-1

Shivlock221b committed:
Level-2

Shivlock221b committed:
Level-2

Shivlock221b committed:
Level-3

Shivlock221b committed:
Level-4

Shivlock221b committed:
Testing UI

Shivlock221b committed:
Tesing UI

Shivlock221b committed:
Level-5

Shivlock221b committed:
Level-6

Shivlock221b committed:
Add save feature

Shivlock221b committed:
Add Save feature

Shivlock221b committed:
date time feature

Shivlock221b committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  date time feature

Shivlock221b committed:
More OOP version

Shivlock221b committed:
Add Package

Shivlock221b committed:
JUnit test

Shivlock221b committed:
Add Javadoc

Shivlock221b committed:
Add JavaDoc

Shivlock221b committed:
Add JavaDoc

Shivlock221b committed:
Feature: find

Shivlock221b committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Add JavaDoc
  Add JavaDoc

Shivlock221b committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Feature: find

Imranr2 commented on src/main/java/src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    /**
+     * method to read from the a text file on the system.
```
Perhaps the header comment should start with a verb? e.g. Reads/Returns/Adds
I have attached the coding standard below
https://se-education.org/guides/conventions/java/intermediate.html
I noticed the same issue in multiple places

Imranr2 commented on src/main/java/src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,132 @@
+     * @param input String input from the user
+     */
+    void addToList(Task input) {
+        //Task task = new Task(input, false);
```
Is this line of code still in use?

Imranr2 commented on src/main/java/src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,36 @@
+    /**
+     * overridden method to give the String representation of the task
+     *
+     * @return
```
Did you forget to include the JavaDocs for the return statement?
I noticed the same issue in multiple places

Imranr2 commented on src/main/java/src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+     */
+    @Override
+    public String toString() {
+        return ("D | " + (isDone ? "1" : "0") + " | " + this.task + " | " + time.format(DateTimeFormatter.ofPattern("MMM d yyyy")));
```
The line is slightly over 120 characters. Perhaps you can split the statement into 2 lines?

Imranr2 commented on src/main/java/src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,8 @@
+package src.main.java.duke;
+
+public class DukeException extends Exception {
```
Perhaps you can include descriptive header comments for the class?
I noticed the same issue in multiple places

Imranr2 commented on src/main/java/src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,38 @@
+/**
+ * Class to represent an event as a task
+ */
+
```
Should there be spaces between documentation block and the method/class?
I believe the following is mentioned in the coding standard

> Punctuation behind each parameter description -No blank line between the documentation block and the method/class

I noticed the same issue in multiple places

Imranr2 reviewed (commented):
LGTM. Just some nits to fix

Shivlock221b committed:
Add Gradle

Shivlock221b committed:
Add GUI

Shivlock221b committed:
Feat: Add assertions

Shivlock221b committed:
Feat: Added update/edit

Shivlock221b committed:
Better Code Quality

Edited Arrow headed code for btter readability
Better comments and header to increase understanding
Added Exceptions for code block issue in deadline time
Removed unnecessary code

Shivlock221b committed:
Add User Guide

Shivlock221b committed:
Merge pull request #2 from Shivlock221b/branch-A-UserGuide

Add User Guide

Shivlock221b committed:
Set theme jekyll-theme-minimal

Shivlock221b committed:
Add CheckStyle

Shivlock221b committed:
Better User Guide

Shivlock221b committed:
Set theme jekyll-theme-slate

Shivlock221b committed:
Set theme jekyll-theme-minimal

Shivlock221b committed:
Set theme jekyll-theme-cayman
